pkg/pmem-csi-driver/nodeserver.go

```diff
@@ -209,14 +236,21 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 		}
 	}
 
-	if req.GetReadonly() {
-		mountOptions = append(mountOptions, "ro")
+	if readOnly {
+		mountFlags = append(mountFlags, "ro")
 	}
 
-	if err := ns.mount(srcPath, targetPath, mountOptions); err != nil {
+	if err := ns.mount(srcPath, targetPath, mountFlags); err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	ns.volReadOnly[req.VolumeId] = readOnly
+	ns.volTargetPath[req.VolumeId] = targetPath
+	// before storing: sort mountFlags slice and join to become single string, as this is what we use to compare
+	askedFlags := req.GetVolumeCapability().GetMount().GetMountFlags()
+	sort.Strings(askedFlags)
+	joinedAskedFlags := strings.Join(askedFlags[:], ",")
```

Comment:
Some code duplication, that we can avoid by reusing the already prepared mount flags array here: https://github.com/intel/pmem-csi/pull/463/commits/fd89169bb0b4c245157c40d554035d0a5353bae8#diff-4a0caa0c2e37db715640b7cbf97d7a1fR212-R214

Reply:
The code above is inside if-block so it is not guaranteed it has created that value.

Reply:
what about we move that computation above the "if" so that it happens always once?

Reply:
> what about we move that computation above the "if" so that it happens always once?

Yes I think, as we always store that info anyway.